### DISCUSSION TITLE
Collection review: cisco.ise

### DISCUSSION THREAD

**Discussion by wastorga** [ansible-collections/ansible-inclusion#46] [In process of inclusion]
Hello, I'd like to submit the cisco.ise collection in its latest version for review:

Collection provides functionality to manage and automate your Cisco Identity Services Engine (ISE) environment with Ansible.

Ansible Galaxy: https://galaxy.ansible.com/cisco/ise
Github: https://github.com/CiscoISE/ansible-ise
Issues Tracker:  https://github.com/CiscoISE/ansible-ise/issues
Github handles: @jbogarin @fmunozmiranda @bvargasre
Is the collection part of Automation Hub: Yes (Waiting for approval)
We meet [Collection Requirements](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst): Yes

**Comment by MarkusTeufelberger** [comment]
`cisco.ise` seems to be already part of Ansible since 5.0.0

**Reply by Andersson007** [reply]
@MarkusTeufelberger thanks for spotting!

6ea5e3a9 (Felix Fontein 2021-10-27 21:10:38 +0200 16) cisco.ise
@felixfontein do you remember the details?

**Reply by felixfontein** [reply]
The application was tracked in https://github.com/ansible-collections/ansible-inclusion/discussions/31, I have no clue why we suddenly have a second application for the same collection. I'll move this to Invalid.

**Comment by fmunozmiranda** [comment]
Hello everyone 

I’d like to kindly request a re-evaluation of the cisco.ise collection for inclusion in Ansible Galaxy / Automation Hub.

The collection was previously marked as inactive due to maintenance issues, but the situation has now changed — the project is fully maintained and actively developed again.

All the issues that led to its exclusion have been resolved, and the collection has been updated and aligned with the latest Ansible standards.

Here are the current references:

GitHub: https://github.com/CiscoISE/ansible-ise

Galaxy: https://galaxy.ansible.com/cisco/ise

Issue Tracker: https://github.com/CiscoISE/ansible-ise/issues

We’d really appreciate if the collection could be reviewed again for inclusion.

Thanks in advance for your time and support! 

— @fmunozmiranda (on behalf of the Cisco ISE Ansible maintainers)

**Reply by felixfontein** [reply]
Background: the collection was determined to be umaintained after not updating their dependency on ansible.utils, which blocked the Ansible 12 release process, and thus got removed from Ansible 12.0.0. Forum thread: https://forum.ansible.com/t/43367

**Reply by mariolenz** [reply]
> I’d like to kindly request a re-evaluation of the cisco.ise collection for inclusion in Ansible Galaxy / Automation Hub.

This inclusion process is about including a collection in the [Ansible Community Package](https://pypi.org/project/ansible/). I'm not sure if you know what this means or really want to do this and follow [our rules](https://docs.ansible.com/ansible/latest/community/collection_contributors/collection_requirements.html).

Maybe you just don't understand the ansible ecosystem and what this inclusion process is about? The collection still _is_ [available on Galaxy](https://galaxy.ansible.com/ui/repo/published/cisco/ise/), even the release you've published yesterday. I don't know much about Automation Hub, but I think being included in ACP isn't a requirement. Maybe you have to be a RH certified collection for this, but I'm just guessing :shrug: 

So please make sure you know what you really want. If you _just_ want your collection available on galaxy: It is anyway. And if you want it to be available on Automation Hub, I don't think being part of ACP is a requirement or helps you there. I might be wrong there, though, because I don't know Automation Hub.

**Reply by Andersson007** [reply]
@fmunozmiranda what do you think about ^ ? You don't need any approval to publish the collection on Galaxy and I also see this collection is already certified on Automation Hub.

**Reply by fmunozmiranda** [reply]
Thanks a lot for the clarification and context 

Yes, our intention is indeed to reapply for inclusion of the cisco.ise collection in the Ansible Community Package (ACP).

We’re aware that the collection was previously removed due to dependency issues with ansible.utils, which caused a release block in Ansible 12. That problem has now been fully addressed — the collection has been updated to comply with the current ansible.utils version and validated with the latest Ansible releases.

**Reply by Andersson007** [reply]
@fmunozmiranda thanks for clarifying and confirming the intention! Once someone from @ansible-collections/steering-committee has time to review it, they'll do, thanks

**Comment by Andersson007** [comment]
# Ansible Collections Checklist (short version)

_For details about the following points, refer to the [Collection Requirements](https://docs.ansible.com/ansible/devel/community/collection_contributors/collection_requirements.html)._

Every comment should say whether the reviewer expects it to be addressed, or whether it's optional.

Note for reviewers: If you don't know how to check any of the points below, please ask maintainers of the collection you're reviewing or a [Steering Committee member](https://docs.ansible.com/ansible/devel/community/steering/community_steering_committee.html#current-steering-committee-members) for clarifications in comments of corresponding inclusion discussion.

**Public availability and communication:**
- [x] published on [Ansible Galaxy](https://galaxy.ansible.com) with version 1.0.0 or later
- [x] have a Code of Conduct (CoC) compatible with the [Ansible Code of Conduct (CoC)](https://docs.ansible.com/ansible/latest/community/code_of_conduct.html)
- [x] have a publicly available issue tracker that does not require a paid level of service to create an account or view issues
- [x] have a public git repository
- [x] releases of the collection are tagged in its repository

**Standards and documentation:**
- [x] adheres to [semantic versioning](https://semver.org/)
- [x] follows [licensing rules](https://docs.ansible.com/ansible/devel/community/collection_contributors/collection_requirements.html#collection-licensing-requirements)
- [x] follows the [Ansible documentation standards](https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_documenting.html) and the [style guide](https://docs.ansible.com/ansible/devel/dev_guide/style_guide/index.html#style-guide)
- [x] follows [development conventions](https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_best_practices.html); as well as these other requirements:
  - [x] modules satisfy the concept of [idempotency](https://docs.ansible.com/ansible/latest/reference_appendices/glossary.html#term-Idempotency>)
  - [x] modules that only gather information are named `<something>_info`
  - [x] modules that return `ansible_facts` are named `<something>_facts` and do not return non-facts
  - [x] other modules must not allow querying information using specific `state` option values, or similar mechanisms (like `state=get` or `state=query`).  These features should be moved to `<something>_info` or `<something>_fact` modules.
  - [x] `check_mode` is supported in all `_info` and `_facts` modules
- [x] supports [all Python versions supported by all ansible-core versions its supports](https://docs.ansible.com/ansible/devel/reference_appendices/release_and_maintenance.html#ansible-core-support-matrix). If it does not, read the [full guidelines](https://docs.ansible.com/ansible/devel/community/collection_contributors/collection_requirements.html#python-compatibility) to see if you qualify for an exception and document the unsupported [Python versions](https://docs.ansible.com/ansible/devel/reference_appendices/release_and_maintenance.html#ansible-core-support-matrix) in the collection ``README.md`` and in every module and plugin (or in doc fragments)
- [x] only uses the [allowed plugin types](https://docs.ansible.com/ansible/devel/community/collection_contributors/collection_requirements.html#modules-plugins) in the `plugins/` directory
- [x] has `README.md`
- [x] FQCNs are used for all plugins and modules including `ansible.builtin.` for builtin ones from ansible-core in all their appearances in the documentation, examples, and return sections, and in `extends_documentation_fragment:`s.
- [x] documentation and return sections use `version_added:` containing the *collection* version for which an option, module or plugin was added (except cases when they were added in the very first release of the collection)
- [x] public plugins, roles and playbooks do not use files outside of `meta/`, `plugins/`, `roles/`, and `playbooks/`

**Collection management:**
- [x] `galaxy.yml` having `tags` field set
- [x] collection dependencies must have a lower bound on the version which is at least 1.0.0, and are all part of the `ansible` package
- [x] `meta/runtime.yml` defines the minimal version of Ansible which the collection works with
- [x] has changelog, preferably with `changelogs/changelog.yaml`
- [x] collection repository should not contain any large objects (binaries) comparatively to the current Galaxy tarball size limit of 20 MB. For example, package installers for testing purposes shouldn't be added.
- [x] collection repository should not contain any unnecessary files like, for example, temporary files. Temporary files should be added to `.gitignore`.

**Tests:**

Note for reviewers: If you don't know how to check the points below, please ask maintainers of the collection you're reviewing how you can do it.
* In most cases, CI is set up via GitHub Actions.
* Check `.yml` files in the `.github/workflows` directory. There must be at least the `sanity` section under `jobs` containing the `ansible-test sanity` command running against all supported ansible-core versions that must be also listed there, for example, `- stable-2.15, - stable-2.16, - stable-2.17`.
* Check workflow runs by clicking the `Actions` tab in the repository's page - you're interested in `Scheduled` runs, runs against release commits and runs against pull requests.
* If there are no workflows in the `Actions` tab, ask the collection maintainers how CI is implemented.

- [x] passed `ansible-test sanity`
- [x] if `test/sanity/ignore*.txt` exists, it MUST not contain error codes listed in the [list of errors that must not be ignored](https://docs.ansible.com/ansible/devel/community/collection_contributors/collection_requirements.html#ci-testing)
- [ ] has CI tests up and running against each of the "major versions" of `ansible-base`/`ansible-core` that the collection supports
  - **MUST FIX** Please add `stable-2.20` to your test matrix. It came our a week ago according to the [core support matrix](https://docs.ansible.com/projects/ansible/devel/reference_appendices/release_and_maintenance.html#ansible-core-support-matrix)
- [x] all CI tests MUST run against every pull request
- [x] all CI tests MUST run regularly (nightly, or at least once per week)
- [x] sanity tests MUST run against a commit that releases the collection; if they don't pass, the collection won't be released

**Reply by Andersson007** [reply]
@wastorga hi, I've found one must fix, please check the list. Thanks

**Reply by fmunozmiranda** [reply]
Fixed.

**Reply by Andersson007** [reply]
@wastorga I approve it, thanks!

**Comment by Andersson007** [comment]
@ansible-collections/steering-committee : Second short review is needed here.
Could anybody please do the second one by checking only the following points (or more if you wish):

- [x] follows the [Ansible documentation standards](https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_documenting.html) and the [style guide](https://docs.ansible.com/ansible/devel/dev_guide/style_guide/index.html#style-guide)
- [x] follows [development conventions](https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_best_practices.html); as well as these other requirements:
  - [x] modules satisfy the concept of [idempotency](https://docs.ansible.com/ansible/latest/reference_appendices/glossary.html#term-Idempotency>)
  - [x] modules that only gather information are named `<something>_info`
  - [x] modules that return `ansible_facts` are named `<something>_facts` and do not return non-facts
  - [x] other modules must not allow querying information using specific `state` option values, or similar mechanisms (like `state=get` or `state=query`).  These features should be moved to `<something>_info` or `<something>_fact` modules.
  - [x] `check_mode` is supported in all `_info` and `_facts` modules

@ansible-collections/steering-committee please conduct the review ^ and explicitly approve the collection.

**Reply by gundalow** [reply]
Approved

**Reply by Andersson007** [reply]
Created a forum topic https://forum.ansible.com/t/default-date-2025-12-03-collection-inclusion-include-cisco-is-in-the-ansible-community-package/44898 FYI

**Comment by Andersson007** [comment]
@wastorga could you please take a look and respond to questions in https://forum.ansible.com/t/default-date-2025-12-03-collection-inclusion-include-cisco-ise-in-the-ansible-community-package/44898/2 ? The inclusion might be postponed / suspended if no response

**Comment by mariolenz** [comment]
For the record: https://forum.ansible.com/t/44898/8
